go/apps/http_api_nostream/resource.py

```diff
@@ -247,6 +247,13 @@ def handle_PUT_in_reply_to(self, request, payload, in_reply_to):
     @inlineCallbacks
     def handle_PUT_send_to(self, request, payload):
         user_account = request.getUser()
+        optout_store = self.worker.vumi_api.get_user_api(user_account) \
+            .optout_store
+        optout = yield optout_store.get_opt_out(
+            payload.get('to_addr_type', 'msisdn'), payload.get('to_addr'))
+        if optout:
+            self.client_error_response(request, "Recipient has opted out")
```

Comment:
Hmm, I'm a bit on the fence here :) Ignoring opted out users when sending messages seems like behaviour the client using the api would expect, so they didn't really make a bad api request. On the other hand, it does seem weird just silently not sending to some of the users that the client asked that messages get sent to. @hodgestar what are your thoughts?